### DISCUSSION TITLE
Convert CCITTG4 to PDF was working but now it isn't working. Missing IM_MOD_RL_CCITTG4_.dll

### DISCUSSION THREAD

**Discussion by ssneed4** [ImageMagick/ImageMagick#7473] [General]
I was using your application to convert CCITTG4 to PDF. Now is missing the IM_MOD_RL_CCITTG4_.dll. Can I get that dll back to perform the conversion?

**Comment by dlemstra** [comment]
We never had that `.dll` so I suspect you are using it differently then you were before.

**Reply by ssneed4** [reply]
You had to have that dll file. Where else would I get the name? That is your naming convention.  I did this through your site before I instlled it I have a batch file that would call your app that I installed. through command line "magick C:\CST\Product_Center\ccittg4\%%b C:\CST\Product_Center\pdfs\%%b.pdf"  In a for loop. Can you check previous versions? This new version is different than what I installed about 2-3 months ago.

**Reply by ssneed4** [reply]
My apologies, the part of going through your site. I did get confused with something else. The part about the missing dll file. That is correct. I converted several files several times using this application.

**Comment by fmw42** [comment]
> magick C:\CST\Product_Center\ccittg4%%b C:\CST\Product_Center\pdfs%%b.pdf

I do not think that is a valid command line command.  We do not use %%b as part of or all the filenames nor as arguments.  

What version of Imagemagick do you think you had used before and what is being used now?  Was your command used via some other tool?

**Reply by ssneed4** [reply]
%%b is just to use in the for loop in the Windows batch file. Pass in the filename + extension .ccittg4. I think it was  ImageMagick-7.1.1-30 vs The current is ImageMagick-7.1.1-34.

**Reply by ssneed4** [reply]
For %%b in (*.*) do (magick C:\CST\Product_Center\ccittg4\%%b C:\CST\Product_Center\pdfs\%%b.pdf)

**Reply by ssneed4** [reply]
For %%b in (*.*) do (magick C:\CST\Product_Center\ccittg4\%%b C:\CST\Product_Center\pdfs\%%b.pdf)

**Reply by ssneed4** [reply]
Weird, it keeps altering the paths. there should be "\" between ccittg4 and %%b and pdfs and %%b.

**Comment by fmw42** [comment]
Have you tried running outside of the loop, just one image, with the actual filenames?  Have you tried uninstalling 7.1.1.34 and reinstall 7.1.1.30?  Does that still work?  Have you changed anything else in your OS?

**Reply by ssneed4** [reply]
I can't find the old version.

**Reply by ssneed4** [reply]
The last time I ran this was on 6-11-2024. 
![image](https://github.com/ImageMagick/ImageMagick/assets/141963601/2b526bf0-cd7f-4413-9097-b7c10fe17e5e)

**Reply by ssneed4** [reply]
Yes, ran it outside of the loop but It is looking for the IM_MOD_RL_CCITTG4_.dll.

**Reply by fmw42** [reply]
Post the exact command (no variables) and the error message that you get

**Reply by ssneed4** [reply]
![image](https://github.com/ImageMagick/ImageMagick/assets/141963601/b5caa3d4-c4ec-4b5e-ace2-fb821edc4900)

**Reply by fmw42** [reply]
Not good form to post images of code and errors.

**Reply by fmw42** [reply]
CCITG4 is a compression type, not an image format.  Post your image to some free hosting service after zipping it, so we can see exactly what image format you are using. Perhaps the suffix got changed.

See https://en.wikipedia.org/wiki/Group_4_compression

**Reply by fmw42** [reply]
P.S.  Imagemagick converts to group4 compression via

magick input.suffix -compress group4 output.suffix

where suffix is some image format that Imagemagick supports with group4, such as TIFF and PDF.

You likely are not recalling your command correctly and started with some other image type and are trying to convert to group4.

See https://imagemagick.org/script/command-line-options.php#compress

**Reply by ssneed4** [reply]
This makes no sense. I had a batch file all setup nothing changed. I converted several of these files back on 6/11/2024 using the windows batch file. Didn't change anything.

**Reply by ssneed4** [reply]
Is there anyway to send me the old version?

**Reply by fmw42** [reply]
One of the IM developers may have it stored somewhere, but I could not find a version online for you.  

But have you read my comments above?  I think you are misremembering what you did before.  

As I said above, CCITG4 is a compression type (called group4 in imagemagick). It is not an image type.  You may have an image such as TIFF or PDF that is compressed in group4 an are trying to convert it to a PDF in group4 compression.  Can you post your script?

And you have not posted an example input image that you think is CCITG4.

**Reply by ssneed4** [reply]
There was nothing to remember. It was all coded and it was tested. Nothing changed, I was waiting for when I needed it.

**Reply by ssneed4** [reply]
I don't know if I can post the files per company policy. These are designs

**Reply by fmw42** [reply]
What tool created these design images?
